File: JAVA/128. Longest Consecutive Sequence.java
class Solution {
    public int longestConsecutive(int[] n) {
        if (n.length == 0)
        return 0;
        Set<Integer> s = new HashSet<>();
        for (int num : n) {
            s.add(num);
        }
        List<Integer> a = new ArrayList<>(s);
        Collections.sort(a);
        int c = 1, m = 1;
        for (int i = 0; i < a.size() - 1; i++) {
            if (a.get(i) + 1 == a.get(i + 1)) 
                c++;
            else
                c = 1;
            m = Math.max(c, m);
        }
        return m;
    }
}

File: C++/128. Longest Consecutive Sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& n) {
        if(n.empty())
        return 0;
        set<int>s(n.begin(),n.end());
        vector<int>a(s.begin(),s.end());
        sort(a.begin(),a.end());
        int c=1,m=1;
        for(int i=0;i<a.size()-1;i++){
            if(a[i]+1==a[i+1])
            c++;
            else
            c=1;
            m=max(c,m);
        }
        return m;
    }
};